webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -28,10 +28,37 @@ import { MessageDuration } from "../message-duration";
 import * as monaco from 'monaco-editor'
 
 let stateCache = {};
+const WHITESPACE_256 = String.fromCharCode.apply(null, new Array(256).fill(0x20));
 let lastFile;
 //Unsupported DS types
 const unsupportedTypes: Array<string> = ['G', 'B', 'C', 'D', 'I', 'R'];
 
+function isContentValidForDataset(content: string[], datasetAttrs: DatasetAttributes) {
```

Comment:
This isn't a big deal, because it has no performance issues, but this is JS syntax btw. We really want,

isContentValidForDataset(content: string[], datasetAttrs: DatasetAttributes): boolean | string {

Reply:
Alright, I will modify this.